src/main/java/com/google/gfx/model/Order.java

```diff
@@ -0,0 +1,78 @@
+package com.google.gfx.model;
+
+import java.time.LocalDateTime;
+import java.util.Currency;
+
+/**
+ * A currency exchange order request to either buy or sell currency.
+ */
+public class Order {
+
+  /**
+   * Order unique identification.
+   */
+  private long id;
+
+  /**
+   * Order type: buy or sell.
+   */
+  private Type type;
+
+  /**
+   * Amount of currency for exchange.
+   */
+  private float amount;
+
+  /**
+   * Currency being bought or sold.
+   */
+  private Currency currency;
+
+  /**
+   * Currency being sought.
+   */
+  private Currency targetCurrency;
```

Comment:
paymentCurrency is also ambiguous, I think.

I still think fromCurrency and toCurrency are pretty self-explanatory. What do you think?